includes/wf_crm_webform_base.inc

```diff
@@ -186,6 +192,41 @@ abstract class wf_crm_webform_base {
   }
 
   /**
+   * Reorder returned results according to settings chosen in wf_civicrm backend
+   *
+   * @param integer $c
+   * @param string $ent
+   * @param array $values
+   * @return array $reorderedArray
+   * @todo Handle user-select case
+   */
+  protected function reorderByLocationType($c, $ent, $values = array()){
+    $reorderedArray = array();
+    // For every entity setting per contact
+    foreach ($this->settings['data']['contact'][$c][$ent] as $setting) {
+      $valueFound = false;
+      foreach($values as $key => $value){
+        if((in_array($ent, array('address', 'email')) && $value['location_type_id'] == $setting['location_type_id'])
+           ||
+          ($value['location_type_id'] == $setting['location_type_id'] && $value[$ent.'_type_id'] == $setting[$ent.'_type_id'])     
```

Comment:
This condition seems to be assuming that `$setting['location_type_id']` is always set, but if the location type is "user select" then this will not be the case :/